libr/asm/p/asm_avr.c

```diff
@@ -15,6 +15,376 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	return op->size = avrdis (op->buf_asm, a->pc, buf, len);
 }
 
+// AVR assembler realization starts
+// by Alexander Bolshev aka @dark_k3y, LGPL -- 2015.
+// heavily based (and using!) on disassemble module
+
+extern instructionInfo instructionSet[AVR_TOTAL_INSTRUCTIONS];
+
+#define MAX_TOKEN_SIZE 8
+#define TOKEN_DELIM " ,\t"
+
+// the next few functions and structures uses for detecting
+// AVR special regs, like X+, -Y, Z+3 in st(d), ld(d) and 
+// (e)lpm instructions
+struct _specialregs {
+	char reg[4];
+	int operandType;
+};
+typedef struct _specialregs specialregs;
+
+#define REGS_TABLE 9 
+
+// radare tolower instruction in rasm, so we use 'x' instead of 'X' etc.
+specialregs RegsTable[REGS_TABLE] = {
+	{"-x", OPERAND_XP}, {"x", OPERAND_X}, {"x+", OPERAND_XP},
+	{"-y", OPERAND_YP}, {"y", OPERAND_Y}, {"y+", OPERAND_YP},
+	{"-z", OPERAND_ZP}, {"z", OPERAND_Z}, {"z+", OPERAND_ZP},
+};
+
+static int parse_specialreg(char *reg){
+	int found = -1;
+	int len = strlen(reg);
```

Comment:
Const char *reg too? :) also i saw some minor indentation bits that must be fixed too. But a new version of llvm is out and maybe ill tey again with clang format to see if now works as we expect in our doc/syntax.md


> On 02 Sep 2015, at 19:46, jvoisin <notifications@github.com> wrote:
> 
> In libr/asm/p/asm_avr.c:
> 
> > +	int operandType;
> > +};
> > +typedef struct _specialregs specialregs;
> > +
> > +#define REGS_TABLE 9 
> > +
> > +// radare tolower instruction in rasm, so we use 'x' instead of 'X' etc.
> > +specialregs RegsTable[REGS_TABLE] = {
> > +	{"-x", OPERAND_XP}, {"x", OPERAND_X}, {"x+", OPERAND_XP},
> > +	{"-y", OPERAND_YP}, {"y", OPERAND_Y}, {"y+", OPERAND_YP},
> > +	{"-z", OPERAND_ZP}, {"z", OPERAND_Z}, {"z+", OPERAND_ZP},
> > +};
> > +
> > +static int parse_specialreg(char *reg){
> > +	int found = -1;
> > +	int len = strlen(reg);
> const int len = strlen (reg); :)
> 
> —
> Reply to this email directly or view it on GitHub.
>